worker/rhpv2.go

```diff
@@ -85,9 +86,12 @@ func (hes HostErrorSet) Error() string {
 	return "\n" + strings.Join(strs, "\n")
 }
 
-func wrapErr(err *error, fnName string) {
+func wrapErr(ctx context.Context, fnName string, err *error) {
 	if *err != nil {
 		*err = fmt.Errorf("%s: %w", fnName, *err)
+		if cause := context.Cause(ctx); cause != nil && !utils.IsErr(*err, cause) {
```

Comment:
Kinda curious what the reason for this is? I'd imagine that usually `err` equals the cause?

Mostly because I'd expect this pattern
```go
select {
  case <-ctx.Done():
    return context.Cause(ctx)
  default:
}
```

Reply:
It's to avoid wrapping the cause twice on an error that has it already. It's just to avoid stuttering error messages in logs.